src/test/shared/sam/cli/samCliLocalInvoke.test.ts

```diff
@@ -0,0 +1,329 @@
+/*!
+ * Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+'use strict'
+
+import * as assert from 'assert'
+import * as del from 'del'
+import * as path from 'path'
+import * as vscode from 'vscode'
+import { writeFile } from '../../../../shared/filesystem'
+import { mkdtemp } from '../../../../shared/filesystemUtilities'
+import { TestLogger } from '../../../../shared/loggerUtils'
+import { DefaultSamCliTaskInvoker } from '../../../../shared/sam/cli/samCliInvoker'
+import { SamCliLocalInvokeInvocation } from '../../../../shared/sam/cli/samCliLocalInvoke'
+
+describe('SamCliLocalInvokeInvocation', async () => {
+
+    class FakeTaskExecution implements vscode.TaskExecution {
+        public constructor(public readonly task: vscode.Task) {
+        }
+
+        public terminate(): void {
+            throw new Error('Method not implemented.')
+        }
+    }
+
+    class TestTaskInvoker implements DefaultSamCliTaskInvoker {
+        public constructor(
+            private readonly onInvoke: (...args: any[]) => void
+        ) {
+        }
+
+        public async invoke(task: vscode.Task): Promise<vscode.TaskExecution> {
+            const shellExecution: vscode.ShellExecution = task.execution as vscode.ShellExecution
+
+            this.onInvoke(shellExecution.args)
+
+            return Promise.resolve(new FakeTaskExecution(task))
+        }
+    }
+
+    let logger: TestLogger
+    let tempFolder: string
+    let placeholderTemplateFile: string
+    let placeholderEventFile: string
+    const nonRelevantArg = 'arg is not of interest to this test'
+
+    before(async () => {
+        logger = await TestLogger.createTestLogger()
+    })
+
+    beforeEach(async () => {
+        tempFolder = await mkdtemp()
+        placeholderTemplateFile = path.join(tempFolder, 'template.yaml')
```

Comment:
This works, but tests will be faster/more stable if we stub out file IO. Optional for this PR, but something to keep in mind moving forward.